minetest/minetest
# CI png optimized check

Adds a workflow to the CI to check whether png files are optimized.
It's only a fuzzy check without `-o7 -zm1-9`, since it would be too slow otherwise, I think, but it should be sufficient.
I hope this makes the review of PRs a little bit easier, and prevents future unoptimized pngs from being added.

Also, it optimizes all existing pngs which have not yet been optimized.
(For some reason, it could save 4.12 KB by optimizing the `cdb_update_cropped.png` file, something most have gone wrong here, since it is only a 16x16 texture.)

## To do

Ready for Review.

## How to test

- Install `optipng` if you don't have it installed
- Run `git ls-files *.png | xargs -I % bash -c "optipng -nc -strip all -simulate % 2>&1 >/dev/null | (if ! grep -q 'is already optimized' $1 ; then echo -e '\033[0;31m%\033[0m'; exit 1; fi)" || (echo "Found non-optimized png file, apply 'optipng -o7 -zm1-9 -nc -strip all -clobber <filename>'"; exit 1;)`
- See how it fails on current master.
